src/smt/command.cpp

```diff
@@ -652,60 +651,58 @@ std::string DeclareSygusFunctionCommand::getCommandName() const
 }
 
 /* -------------------------------------------------------------------------- */
-/* class SynthFunCommand                                                    */
+/* class SynthFunCommand                                                      */
 /* -------------------------------------------------------------------------- */
 
-SynthFunCommand::SynthFunCommand(const std::string& id,
-                                 Expr func,
-                                 Type sygusType,
+SynthFunCommand::SynthFunCommand(const api::Solver* solver,
+                                 const std::string& id,
+                                 const std::vector<api::Term>& vars,
+                                 api::Sort sort,
                                  bool isInv,
-                                 const std::vector<Expr>& vars)
+                                 std::unique_ptr<api::Grammar> g)
```

Comment:
I believe this should take `Grammar * g` not `std::unique_ptr<api::Grammar> g`.

The point is that the command does not own the grammar (which is owned by the user of this command).

Reply:
Done!